test/e2e/validation_test.go

```diff
@@ -165,37 +162,49 @@ func validateInstanceSecret(kubeclient kubernetes.Interface, namespace, instance
 			return false, err
 		}
 		encodedCreds := instanceSecret.Data[instanceID]
-
-		var creds tracker.Credentials
 		if err := json.Unmarshal(encodedCreds, &creds); err != nil {
 			return false, errors.Wrap(err, "unmarshalling creds failed")
 		}
-
-		if err := validateConnectivity(ipAddress, instanceID, creds); err == nil {
-			log.Println("Successfully validated the SSH Connection")
-			return true, nil
-		}
-
-		log.Printf("failed with error for creds %v: %v", creds, err)
-		return false, nil
+		return true, nil
 	})
+	return creds, err
+}
+
+// validateInstanceSecret validates the instance secret.
+func (t *testContext) validateInstanceSecret(instanceID string) error {
+	ipAddress, err := t.getInstanceIP(instanceID)
+	if err != nil {
+		return err
+	}
+	creds, err := t.getCredsSecretData(instanceID)
+	if err != nil {
+		return err
+	}
+	if creds == (tracker.Credentials{}) {
+		return errors.New("expected credentials to be present but got a nil value")
+	}
+	windowsVM, err := t.getWindowsVM(ipAddress, instanceID, creds)
+	if err != nil {
+		return err
+	}
+	// This may not be needed but if you want to store the slice of Windows VMs.
+	t.windowsVMs = append(t.windowsVMs, windowsVM)
+	if err := t.validateConnectivity(windowsVM); err == nil {
```

Comment:
I'm confused, whats the point of this? Its not a test

Reply:
Good point. This needs an assert.

Reply:
This can't be an assert, considering it's not a test suite. I did not understand what you mean exactly by it's not a test. This is supposed to provide some validation for the credentials